lib/worker.js

```diff
@@ -84,30 +80,78 @@ async function getLocalLinter(basedir) {
             linter = require('loophole').allowUnsafeNewFunction(() => require(linterPath).Linter);
           }
         } else {
-          linter = tslintDef;
+          return resolve();
         }
-        tslintCache.set(basedir, linter);
+        tslintCache.set(fileDir, linter);
         return resolve(linter);
       },
     );
   });
 }
 
+function getNodePrefixPath() {
+  return new Promise((fulfill, reject) => {
+    const npmCommand = process.platform === 'win32' ? 'npm.cmd' : 'npm';
+    ChildProcess.exec(
+      `${npmCommand} get prefix`,
+      { env: Object.assign(Object.assign({}, process.env), { PATH: getPath() }) },
+      (err, stdout, stderr) => {
+        if (err || stderr) {
+          reject(err || new Error(stderr));
+        } else {
+          fulfill(stdout.trim());
+        }
+      },
+    );
+  });
+}
+
 async function getLinter(filePath) {
   const basedir = path.dirname(filePath);
   if (tslintCache.has(basedir)) {
     return tslintCache.get(basedir);
   }
 
-  // Initialize the default instance if it hasn't already been initialized
-  loadDefaultTSLint();
-
   if (config.useLocalTslint) {
-    return getLocalLinter(basedir);
+    const localLint = await resolveAndCacheLinter(basedir);
+    if (localLint) {
+      return localLint;
+    }
+  }
+
+  if (fallbackLinter) {
+    tslintCache.set(basedir, fallbackLinter);
+    return fallbackLinter;
+  }
+
+  if (config.useGlobalTslint) {
+    if (config.globalNodePath) {
+      const globalLint = await resolveAndCacheLinter(basedir, config.globalNodePath);
+      if (globalLint) {
+        fallbackLinter = globalLint;
+        return globalLint;
+      }
+    }
+
+    const globalLint = await getNodePrefixPath()
+      .then(prefix => resolveAndCacheLinter(basedir, prefix))
+      .catch((err) => {
+        // eslint-disable-next-line no-console
+        console.warn(`Attempted to load global tslint, but \`npm get prefix\`
+failed. Falling back to the packaged version of tslint. You can specify
+your prefix manually in the settings or linter-tslint config file.\n
+The error message encountered was:\n\n${err.message}`);
+      });
+    if (globalLint) {
+      fallbackLinter = globalLint;
```

Comment:
❓ Should this be cached in `tslintCache`?

Reply:
I would probably leave `fallbackLinter` in its own variable, personally? I mean, you *could* put it into the map with everything else, but since it could technically be the global linter *or* the linter-tslint packaged version, that sorta makes finding it again a little complicated. This way, it's just "No local linter? Alright, use the other one." and you're done.

Reply:
Makes sense. 👍